core/src/core/transaction.rs

```diff
@@ -176,6 +285,19 @@ impl KernelFeatures {
 				let lock_height = reader.read_u64()?;
 				KernelFeatures::HeightLocked { fee, lock_height }
 			}
+			KernelFeatures::NO_RECENT_DUPLICATE_U8 => {
+				// NRD kernels are invalid if NRD feature flag is not enabled.
+				if !global::is_nrd_enabled() {
+					return Err(ser::Error::CorruptedData);
```

Comment:
what happened with error NRDKernelNotEnabled ?

Reply:
We don't have a single global set of errors.

This code needs to return a serialization error, and we use the pattern of returning a relatively generic `CorruptedData` for serialization errors like this.
`NRDKernelNotEnabled` is a block pipeline error and cannot be used here.